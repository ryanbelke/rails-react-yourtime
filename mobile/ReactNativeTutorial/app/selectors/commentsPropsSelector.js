import { Map } from 'immutable';
import { createSelector } from 'reselect';
import commentsStoreSelector from './commentsStoreSelector';

export default createSelector(
  commentsStoreSelector,
  commentsStore => Map({
    comments: commentsStore.delete('meta').valueSeq().sortBy(x => x.get('id')).reverse(),
    meta: commentsStore.get('meta'),
  })
);
