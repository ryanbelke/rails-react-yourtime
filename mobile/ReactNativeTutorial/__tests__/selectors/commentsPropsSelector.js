import { fromJS } from 'immutable';
import commentsPropsSelector from 'ReactNativeTutorial/app/selectors/commentsPropsSelector';

describe('commentsPropsSelector', () => {
  it('maps commentsStore to props', () => {
    const state = fromJS({
      commentsStore: {
        2: { id: 2, author: 'Justin', text: 'Another random comment' },
        3: { id: 3, author: 'John', text: 'Yet another random comment' },
        1: { id: 1, author: 'Alexey', text: 'Just some random comment' },
        meta: { loading: false },
      },
    });
    expect(commentsPropsSelector(state)).toMatchSnapshot();
  });
});
