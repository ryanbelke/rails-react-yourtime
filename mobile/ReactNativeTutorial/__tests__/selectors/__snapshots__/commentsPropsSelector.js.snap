exports[`commentsPropsSelector maps commentsStore to props 1`] = `
Object {
  "comments": Array [
    Object {
      "author": "John",
      "id": 3,
      "text": "Yet another random comment",
    },
    Object {
      "author": "Justin",
      "id": 2,
      "text": "Another random comment",
    },
    Object {
      "author": "Alexey",
      "id": 1,
      "text": "Just some random comment",
    },
  ],
  "meta": Object {
    "loading": false,
  },
}
`;
